common/src/main/java/fudge/notenoughcrashes/stacktrace/ModIdentifier.java

```diff
@@ -110,6 +132,42 @@ private static Set<CommonModMetadata> identifyFromClass(String className, ModsBy
         }
     }
 
+    @Nullable
+    private static MixinMerged findMixinMerged(StackTraceElement element) {
+        try {
+            Class<?> clazz = Class.forName(element.getClassName());
+            // Walk through methods because we don't know parameter types
+            Method[] methods = clazz.getDeclaredMethods();
+            for (Method method : methods) {
+                if (method.getName().equals(element.getMethodName())) {
+                    MixinMerged mixinMerged = method.getAnnotation(MixinMerged.class);
+                    if (mixinMerged != null) {
+                        return mixinMerged;
+                    }
+                }
+            }
+        } catch (ClassNotFoundException ignored) {}
+
+        return null;
+    }
+
+    @NotNull
+    private static Set<CommonModMetadata> identifyFromMixin(IMixinInfo mixin) {
+        Set<CommonModMetadata> metadata = mixinConfigToMods.get(mixin.getConfig());
+        if (metadata != null) return metadata;
+
+        String mixinFileName = mixin.getConfig().getName();
+        Set<CommonModMetadata> modsWithMixinFile = new LinkedHashSet<>();
+        for (CommonModMetadata mod : NecPlatform.instance().getAllMods()) {
+            Path mixinFile = mod.rootPath().resolve(mixinFileName);
+            if (Files.exists(mixinFile)) {
+                modsWithMixinFile.add(mod);
```

Comment:
Is it possible for multiple mods to have the same mixin file name? If not, modsWithMixinFile shouldn't be a set.

Reply:
Multiple mods could have the same mixin file name, at least under Fabric/Quilt, since they're each treated as individual filesystems. I could parse it with GSON or something to check if the mixin is specifically in that given file